ding/world_model/mbpo.py

```diff
@@ -0,0 +1,258 @@
+import itertools
+import numpy as np
+import copy
+import torch
+from torch import nn
+
+from ding.utils import WORLD_MODEL_REGISTRY
+from ding.utils.data import default_collate
+from ding.world_model.base_world_model import HybridWorldModel
+from ding.world_model.model.ensemble import EnsembleModel, StandardScaler
+
+
+@WORLD_MODEL_REGISTRY.register('mbpo')
+class MBPOWorldModel(HybridWorldModel, nn.Module):
+    config = dict(
+        model=dict(
+            network_size=7,
+            elite_size=5,
+            state_size=None,
+            action_size=None,
+            reward_size=1,
+            hidden_size=200,
+            use_decay=False,
+            batch_size=256,
+            holdout_ratio=0.2,
+            max_epochs_since_update=5,
+            deterministic_rollout=True,
+        ),
+    )
+
+    def __init__(self, cfg, env, tb_logger):
+        HybridWorldModel.__init__(self, cfg, env, tb_logger)
+        nn.Module.__init__(self)
+
+        cfg = cfg.model
+        self.network_size = cfg.network_size
+        self.elite_size = cfg.elite_size
+        self.state_size = cfg.state_size
+        self.action_size = cfg.action_size
+        self.reward_size = cfg.reward_size
+        self.hidden_size = cfg.hidden_size
+        self.use_decay = cfg.use_decay
+        self.batch_size = cfg.batch_size
+        self.holdout_ratio = cfg.holdout_ratio
+        self.max_epochs_since_update = cfg.max_epochs_since_update
+        self.deterministic_rollout = cfg.deterministic_rollout
+
+        self.ensemble_model = EnsembleModel(
+            self.state_size,
+            self.action_size,
+            self.reward_size,
+            self.network_size,
+            self.hidden_size,
+            use_decay=self.use_decay
+        )
+        self.scaler = StandardScaler(self.state_size + self.action_size)
+
+        if self._cuda:
```

Comment:
maybe `self.device` can be more general, you can use `tensor.to(self.device)`

Reply:
Using boolean `cuda` is more consistent with the practice in `ding.policy`